src/core/search.rs

```diff
@@ -246,14 +246,25 @@ pub trait FindexSearch<
                 // Use a vector not to shuffle the chain. This is important because indexed
                 // values can be divided in blocks that span several lines in the chain.
                 for (uid, value) in encrypted_item {
-                    chain.push((
-                        uid,
-                        ChainTableValue::<BLOCK_LENGTH>::decrypt::<
-                            TABLE_WIDTH,
-                            DEM_KEY_LENGTH,
-                            DemScheme,
-                        >(&kwi_value, &value)?,
-                    ));
+                    let decrypted_value = ChainTableValue::<BLOCK_LENGTH>::decrypt::<
+                        TABLE_WIDTH,
+                        DEM_KEY_LENGTH,
+                        DemScheme,
+                    >(&kwi_value, &value)
+                    .map_err(|_| {
+                        FindexErr::CallBack(format!(
+                            "fail to decrypt one of the `value` returned by the fetch chains \
+                             callback (uid as hex was '{}', value {})",
+                            hex::encode(&uid),
+                            if value.is_empty() {
+                                "was empty".to_owned()
```

Comment:
I think `to_string` is more explicit. I think I may not be the only one to prefer this since I saw some change in other PR about this. Maybe ask around and ignore it if I am the only one.

```suggestion
                                "was empty".to_string()
```

Reply:
> You should always be using to_owned(). to_string() is the generic conversion to a String from any type implementing the ToString trait. It uses the formatting functions and therefor might end up doing multiple allocations and running much more code than a simple to_owned() which just allocates a buffer and copies the literal into the buffer.

`to_owned()` is faster since it's a specific implementation for `String` instead of a generic one (and I think it also improves compile-time since it doesn't involves traits)

Reply:
Then could you change all occurrences in the code base ?

Reply:
Not here, not the point of this MR, we can do this step by step when we encounter some `to_string()` that could be replaced. It's not a priority (the gains are minimal I think)